### PR TITLE
Exposing the node capacity via chitchat

### PR DESCRIPTION
**Pull request by fulmicoton**

### CODE REVIEW

File: quickwit/quickwit-config/src/node_config/mod.rs [hunk 1]
```diff
@@ -55,6 +55,8 @@ pub struct IndexerConfig {
     pub enable_otlp_endpoint: bool,
     #[serde(default = "IndexerConfig::default_enable_cooperative_indexing")]
     pub enable_cooperative_indexing: bool,
+    #[serde(default = "IndexerConfig::default_capacity")]
+    pub capacity: u32,
```
Comment (guilload) [inline quickwit/quickwit-config/src/node_config/mod.rs:59]: I think `capacity` is too ambiguous. Also, I'd rather expose our users to the pipeline concept as little as possible. I think we should ask our users how many CPUs they want to allocate to their indexers, and we infer how many pipelines we can pack from there. With that in mind, `cpu_capacity` is probably what we want.

https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#resource-units-in-kubernetes

Comment (fulmicoton) [reply quickwit/quickwit-config/src/node_config/mod.rs]: You are right. I introduced a CpuCapacity. We can now configure it either as a string "1000m" or as a float. (1.0)

File: quickwit/quickwit-proto/src/indexing/mod.rs [hunk 2]
```diff
@@ -209,13 +210,146 @@ impl TryFrom<&str> for IndexingTask {
...
+        let milli_cpus: u32 = milli_cpus_without_unit_str
+            .parse::<u32>()
+            .map_err(|_err| format!("invalid cpu capacity: `{cpu_capacity_str}`."))?;
+        Ok(CpuCapacity(milli_cpus))
+    }
+}
+
+impl From<CpuCapacity> for CpuCapacityForSerialization {
+    fn from(cpu_capacity: CpuCapacity) -> CpuCapacityForSerialization {
+        CpuCapacityForSerialization::MilliCpuWithUnit(format!("{}m", cpu_capacity.0))
```
Comment (guilload) [inline quickwit/quickwit-proto/src/indexing/mod.rs:352]: I know it's tested in `quickwit-config`, but this could be tested in this file directly.

**Review by guilload — APPROVED** [review]
nit: `CpuCapacity` is also sometimes used for measuring a load or a threshold. It might be better to use `CpuUnits` for the type, and let the variable names tell us whether it's a capacity, load, etc.

File: quickwit/quickwit-proto/src/indexing/mod.rs (hunk 2 above)
Comment (fulmicoton) [reply quickwit/quickwit-proto/src/indexing/mod.rs:352]: adding tests

**Pull request merged**